lib/restrictedsession/restricted_test.go

```diff
@@ -139,83 +140,50 @@ type Suite struct {
 	expectedAuditEvents []apievents.AuditEvent
 }
 
-var _ = check.Suite(&Suite{})
+func setupBPFContext(t *testing.T) *bpfContext {
+	tt := bpfContext{}
 
-func TestRootRestrictedSession(t *testing.T) {
-	if !bpfTestEnabled() {
-		t.Skip("BPF testing is disabled")
-	}
-
-	check.TestingT(t)
-}
-
-func mustParseIPSpec(cidr string) *net.IPNet {
-	ipnet, err := ParseIPSpec(cidr)
-	if err != nil {
-		panic(err)
-	}
-	return ipnet
-}
-
-type mockClient struct {
-	restrictions api.NetworkRestrictions
-	services.Fanout
-}
-
-func (mc *mockClient) GetNetworkRestrictions(context.Context) (api.NetworkRestrictions, error) {
-	return mc.restrictions, nil
-}
-
-func (_ *mockClient) SetNetworkRestrictions(context.Context, api.NetworkRestrictions) error {
-	return nil
-}
-
-func (_ *mockClient) DeleteNetworkRestrictions(context.Context) error {
-	return nil
-}
-
-func (s *Suite) SetUpSuite(c *check.C) {
 	utils.InitLoggerForTests()
 
 	// This test must be run as root and the host has to be capable of running
 	// BPF programs.
 	if !isRoot() {
-		c.Skip("Tests for package restrictedsession can only be run as root.")
+		t.Skip("Tests for package restrictedsession can only be run as root.")
 	}
 
-	s.srcAddrs = map[int]string{
+	tt.srcAddrs = map[int]string{
 		4: "0.0.0.0",
 		6: "::",
 	}
 
 	var err error
 	// Create temporary directory where cgroup2 hierarchy will be mounted.
-	s.cgroupDir, err = os.MkdirTemp("", "cgroup-test")
-	c.Assert(err, check.IsNil)
+	tt.cgroupDir, err = os.MkdirTemp("", "cgroup-test")
```

Comment:
Is anything going to clean up this temp dir? Should you use `t.TempDir()` instead so you don't have to worry about it?